src/StatusQ/Layout/StatusAppLayout.qml

```diff
@@ -1,49 +1,88 @@
 import QtQuick 2.13
-import QtQuick.Layouts 1.13
+import QtQuick.Layouts 1.14
 import QtQuick.Controls 2.13
 
-import StatusQ.Core 0.1
+import StatusQ.Components 0.1
 import StatusQ.Core.Theme 0.1
 
-Rectangle {
-    id: statusAppLayout
-
-    implicitWidth: 900
+SplitView {
+    id: root
+    implicitWidth: 822
     implicitHeight: 600
 
-    color: Theme.palette.statusAppLayout.backgroundColor
-    
-    property StatusAppNavBar appNavBar
-    property Item appView
+    handle: Item { }
+
+    property Item headerContent
+    property Item leftPanel
+    property Item centerPanel
+    property Component rightPanel
+
+    property bool showRightPanel
+    property bool showHeader: true
+
+    property int notificationCount
+    property alias headerContent: statusToolBar.headerContent
+    property alias notificationButton: statusToolBar.notificationButton
+
+    signal notificationButtonClicked()
 
-    onAppNavBarChanged: {
-        if (!!appNavBar) {
-            appNavBar.parent = appNavBarSlot
+    onCenterPanelChanged: {
+        if (!!centerPanel) {
+            centerPanel.parent = centerPanelSlot;
         }
     }
 
-    onAppViewChanged: {
-        if (!!appView) {
-            appView.parent = appViewSlot
+    Control {
+        SplitView.minimumWidth: (!!leftPanel) ? 300 : 0
+        SplitView.preferredWidth: (!!leftPanel) ? 300 : 0
+        SplitView.fillHeight: (!!leftPanel)
+        background: Rectangle {
+            anchors.fill: parent
+            color: Theme.palette.baseColor4
         }
+        contentItem: (!!leftPanel) ? leftPanel : null
     }
 
-    Row {
-        id: rowLayout
-        spacing: 0
-        height: statusAppLayout.height
-        width: statusAppLayout.width
-
-        Item {
-            id: appNavBarSlot
-            height: statusAppLayout.height
-            width: 78
+    Control {
+        SplitView.minimumWidth: (!!centerPanel) ? 300 : 0
+        SplitView.fillWidth: (!!centerPanel)
+        SplitView.fillHeight: (!!centerPanel)
+        leftPadding: (!!centerPanel) ? -5 : 0
+        background: Rectangle {
+            anchors.fill: parent
+            color: Theme.palette.statusAppLayout.rightPanelBackgroundColor
+        }
+        contentItem: Item {
+            anchors.fill: parent
+            StatusToolBar {
```

Comment:
This should be the central part imho, not the parent contentItem and parent Control; (Status)ToolBar should really be a QQC2.Control

Reply:
it's the central component, what do you mean exactly?

Reply:
I mean, instead of the Control at line 131, why not make it StatusToolBar, and customize its background/contentItem? Would make the code a lot simpler.